server/controllers/stock/core.js

```diff
@@ -259,3 +271,231 @@ function stockManagementProcess(inventories) {
+    JOIN inventory i ON i.uuid = l.inventory_uuid
+    JOIN inventory_unit iu ON iu.id = i.unit_id 
+    JOIN depot d ON d.uuid = m.depot_uuid 
+  `;
+
+  const clause = ' GROUP BY l.inventory_uuid, m.depot_uuid ';
+
+  return getLots(sql, params, clause)
+    .then((rows) => {
+      bundle.inventories = rows;
+      return getStockConsumptionAverage(null, params.dateTo);
+    })
+    .then((rows) => {
+      var sameInventory;
+      var sameDepot;
+
+      bundle.consumptions = rows;
+
+      for (let i = 0; i < bundle.consumptions.length; i++) {
+        for (let j = 0; j < bundle.inventories.length; j++) {
+          sameInventory = bundle.consumptions[i].uuid === bundle.inventories[j].inventory_uuid;
+          sameDepot = bundle.consumptions[i].depot_uuid === bundle.inventories[j].depot_uuid;
+          if (sameInventory && sameDepot) {
+            bundle.inventories[j].avg_consumption = bundle.consumptions[i].quantity;
+            break;
+          }
+        }
+      }
+
+      return bundle.inventories;
+    })
+    .then((inventories) => stockManagementProcess(inventories, delay, purchaseInterval))
+    .then((rows) => {
+      if (status) {
+        return rows.filter(row => row.status === status);
+      }
+      return rows;
+    });
+}
+
+/**
+ * Inventory Movement Report
+ */
+function getInventoryMovements(params) {
+  const bundle = {};
+
+  const sql = `
+    SELECT BUID(l.uuid) AS uuid, l.label, l.initial_quantity,
+        d.text AS depot_text, l.unit_cost, l.expiration_date,
+        m.quantity, m.is_exit, m.date,
+        BUID(l.inventory_uuid) AS inventory_uuid, BUID(l.origin_uuid) AS origin_uuid, 
+        l.entry_date, i.code, i.text, BUID(m.depot_uuid) AS depot_uuid,
+        i.avg_consumption, i.purchase_interval, i.delay, iu.text AS unit_type
+    FROM stock_movement m 
+    JOIN lot l ON l.uuid = m.lot_uuid
+    JOIN inventory i ON i.uuid = l.inventory_uuid
+    JOIN inventory_unit iu ON iu.id = i.unit_id 
+    JOIN depot d ON d.uuid = m.depot_uuid 
+  `;
+
+  return getLots(sql, params, ' ORDER BY m.date ASC ')
+    .then((rows) => {
+      bundle.movements = rows;
+
+      // build the inventory report
+      let stockQuantity = 0;
+      let stockUnitCost = 0;
+      let stockValue = 0;
+
+      // stock method CUMP : cout unitaire moyen pondere
+      const movements = bundle.movements.map((line) => {
+        const movement = {
+          date : line.date,
+          entry : { quantity : 0, unit_cost : 0, value : 0 },
+          exit : { quantity : 0, unit_cost : 0, value : 0 },
+          stock : { quantity : 0, unit_cost : 0, value : 0 },
+        };
+
+        if (line.is_exit) {
+          stockQuantity -= line.quantity;
+          stockValue = stockQuantity * stockUnitCost;
+
+          // exit
+          movement.exit.quantity = line.quantity;
+          movement.exit.unit_cost = stockUnitCost;
+          movement.exit.value = line.quantity * line.unit_cost;
+
+          // stock status
```

Comment:
It looks like you have repeated these lines of code (after `// stock status`).  Could you move those out of the `if/else` block?  It would make the code easier to read to only putting logic that depends on the `if/else` in the `if/else` block.